packages/core-mobile/app/contexts/DeeplinkContext/DeeplinkContext.tsx

```diff
@@ -39,78 +40,95 @@ export const DeeplinkContextProvider = ({
   children: React.ReactNode
 }): JSX.Element => {
   const dispatch = useDispatch()
+  const store = useStore()
   const walletState = useSelector(selectWalletState)
   const isWalletActive = walletState === WalletState.ACTIVE
   const isDeveloperMode = useSelector(selectIsDeveloperMode)
   const isNotificationBlocked = useSelector(selectIsNotificationBlocked)
   const processedFeatureFlags = useSelector(selectFeatureFlags)
   const [pendingDeepLink, setPendingDeepLink] = useState<DeepLink>()
 
-  const expireDeepLink = useCallback(() => {
-    setPendingDeepLink(undefined)
-  }, [])
+  const maybeSetActiveNetwork = useCallback(
+    (data: NotificationData) => {
+      if (
+        'chainId' in data &&
+        ['string' || 'number'].includes(typeof data.chainId)
+      ) {
+        const chainId = Number(data.chainId)
+        const network = selectNetwork(chainId)(store.getState() as RootState)
```

Comment:
have you tried using getNetwork() instead? using the store object here is kinda scary.

Reply:
[Use useNetworks](https://github.com/ava-labs/core-mobile/pull/1659/commits/fe9c230358329a8ab2e725cb032cecc3115d89ca)